enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,481 @@
+ip-10-0-59-56.ec2.internal    True      False             False              False                    False              False
+ip-10-0-6-214.ec2.internal    True      False             False              False                    False              False
+```
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARED   UPDATEEXECUTED   UPDATEPOSTACTIONCOMPLETED   UPDATECOMPLETED   RESUMED
+ip-10-0-12-194.ec2.internal   False     True              True               True                     True              Unknown
+ip-10-0-17-102.ec2.internal   True      False             False              False                    False             False
+ip-10-0-2-232.ec2.internal    True      False             False              False                    False             False
+ip-10-0-59-251.ec2.internal   True      False             False              False                    False             False
+ip-10-0-59-56.ec2.internal    True      False             False              False                    False             False
+ip-10-0-6-214.ec2.internal    True      False             False              False                    False             False
+```
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARED   UPDATEEXECUTED   UPDATEPOSTACTIONCOMPLETED   UPDATECOMPLETED   RESUMED
+ip-10-0-12-194.ec2.internal   False     True              True               True                     True              True
+ip-10-0-17-102.ec2.internal   True      False             False              False                    False             False
+ip-10-0-2-232.ec2.internal    True      False             False              False                    False             False
+ip-10-0-59-251.ec2.internal   True      False             False              False                    False             False
+ip-10-0-59-56.ec2.internal    True      False             False              False                    False             False
+ip-10-0-6-214.ec2.internal    True      False             False              False                    False             False
+```
+
+The general process here goes as follows:
+
+A state (and its respective child states) will generally go from False -> Unkown -> True -> False.
+The states are mostly in the past tense. This is because processes like `Drained` are defined by completion primarily not the progress. So a user will have `UpdateExecuted` == Unknown and `Drained` == Unkown until the Drain actually completes.
+However, the unknown phase will be accompanied by a message for how the drain is currently going or if the drain has gone wrong.
+
+The condition transitions back to False once the update process is completed. Once Updated == true is the case, all previous states get set to false and their reasons/messages show the fact that this was the message from the previous update cycle.
```

Comment:
the messages currently look like this 

```
    Last Transition Time:  2024-01-12T14:52:19Z
    Message:               Action during update to rendered-master-eef3a1a1a422b7ebd085b346042ca5a8: Upgrade required a reboot. Completed this as the post update action.
    Reason:                UpdatePostActionCompleteRebootedNode
    Status:                False
    Type:                  RebootedNode
```

is that alright? we can create a story/bug to change this. But for the EP, this is what merged

Reply:
the action is reset when the next update gets to this phase. I can make it so that they all get reset when an update begins? I can make a card for this.